pkg/asset/installconfig/azure/validation.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	aznetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+
+	"github.com/openshift/installer/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Validate executes platform-specific validation.
+func Validate(client API, ic *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("azure"))...)
+	return allErrs.ToAggregate()
+}
+
+// validateNetworks checks that the user-provided VNet is in the project and the provided subnets are valid.
+func validateNetworks(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
```

Comment:
please pass in explicit arguments like the Azure.Platform rather than complete install-config here.

Reply:
I don't disagree with you here, however, this is the same as GCP.

Reply:
hmm.

we should fix those too, later on.

Reply:
Looking at this more closely, what is the issue with passing the installconfig? If you look at validation_test.go, a valid install config is necessary for validation. This seems the easiest approach and makes the most sense. Breaking it into several parameters (azure, machine cidr, etc) seems more complicated to me.

Reply:
> Looking at this more closely, what is the issue with passing the installconfig?

because you are trying to validate the network, but you need the entire install-config for that?
most probably, not. and explicitly input arguments make it clear to the reader what is required.